WPMedia/fusion-news-theme-blocks
# TMEDIA-454:  usePaywall

## Description
- A large chunk of the logic for this was pulled from the following PR: https://github.com/WPMedia/fusion-news-theme-blocks/pull/1045/files
- I feel as though some of this **might** change come implementation, but I took the POC and cleaned it up and provided testing to the different branches of code.

## Jira Ticket
- [TMEDIA-454](https://arcpublishing.atlassian.net/browse/TMEDIA-454)

## Acceptance Criteria
* Output type renders paywall script if paywall src is provided in blocks.json
  * src - would be a URL either absolute or relative
  * output type is tested
* Paywall hook sets up paywall running logic, passing proper options to `ArcP.run`
  * consumes useIdentity hook. (** edit: Nick ** Consumes Identity per POC)
  * takes globalContent, runs ArcP.run for any new content ID (canonical_url?) & user
  * returns {isPaywalled, paywallType (regwall/paywall), campaignCode};  (** edit: Nick ** returns information in a different form per POC)
* Unit tests exist for usePaywall hook

## Test Steps
1. Checkout this branch `git checkout TMEDIA-454-usePaywall`
2. Run test: `blocks/subscriptions-block/components/usePaywall.test.jsx`
3. Run `npx fusion start -f -l @wpmedia/default-output-block`
4. Verify script tag is located in header for the specific site (http://localhost/pf/all-blocks/?_website=the-gazette): 
![image](https://user-images.githubusercontent.com/2287238/136254066-14993aff-b4c5-4354-a915-27090fde0393.png)

## Dependencies or Side Effects
Depends on blocks.json properties created in: https://github.com/WPMedia/Fusion-News-Theme/pull/196
Depends on firewall rules being made: https://corecomponents.arcpublishing.com/subscriptions/paywall/rules/new?site=the-gazette (eg: gazette site)

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
